src/main/resources/api.yaml

```diff
@@ -235,3 +298,22 @@ definitions:
     required:
       - enrollmentPeriod
       - durationPeriod
+
+  AvailabilityDto:
+    type: object
+    properties:
+      userId:
+        type: integer
+        format: int64
+      timeAvailable:
+        type: integer
+        description: In minutes
+      timeRemaining:
+        type: integer
+        description: In minutes
+      notes:
+        type: string
+    required:
+      - userId
```

Comment:
A jak robimy PUT to też wymagamy `userId`?

Reply:
No i właśnie w taki sposób się objawia używanie jednej klasy w kilku miejscach. Bo z jednej strony musi być w przypadku pobierania listy dostępności wszystkich userów, a z drugiej nie musi przy zarządzaniu swoją dostępnością przez konkretnego usera

Reply:
Ja nie requestowałem obecności `userId`, nie potrzebne to przy liście userów. Propozycja która się pojawiła była taka, że `userId` może się gdzieś indziej przyda to może już teraz dodajmy. Jak tutaj to zawadza to chyba można usunąć

Reply:
Ja nie widzę potrzeby - w wielu miejscach używamy tego samego typu obiektu w różnych celach - teraz `required` Swaggerowe nam się rozjeżdża i dochodzi walidacja, że `userId` nie jest nullem w konkretnym casie - gdzie i tak trzeba sprawdzić sto innych rzeczy (projekt istnieje, sprint w projekcie istnieje, user jest w projekcie...). 

`UserAvailabilityDto` i `AvailabilityDto` nie są dla mnie dobrymi nazwami obok siebie, ale jak ktoś ma inne propozycje, to spoko

Reply:
Proponuję zostać przy tym co już jest xd